src/ncurses_ui.cc

```diff
@@ -661,12 +661,27 @@ Optional<Key> NCursesUI::get_next_key()
         const int new_c = wgetch(m_window);
         if (new_c == '[') // potential CSI
         {
-            const Codepoint csi_val = wgetch(m_window);
-            switch (csi_val)
-            {
-                case 'I': return {Key::FocusIn};
-                case 'O': return {Key::FocusOut};
-                default: break; // nothing
+            int keycode = 0;
+            Codepoint csi_val;
+
+            while ((csi_val = wgetch(m_window)) != ERR) {
+                if ('0' <= csi_val && csi_val <= '9')
```

Comment:
`if ('0' >= csi_val && csi_val <= '9')` ?

Reply:
That condition would be the same as `csi_val <= '0' && csi_val <= '9'`, and since `'0' < '9'`, that would be the same as `csi_val < '0'`, which isn't strict enough to match ASCII digits.

Reply:
`std::isdigit()` seems to accept only ASCII digits, anyway.  Unless in Microsoft 1252 codepage on Microsoft compilers, but we only support UTF-8.

Reply:
(Link https://en.cppreference.com/w/cpp/string/byte/isdigit)

Reply:
And... ignore all that, because it's a Codepoint.  Ne'ermind.

Reply:
I misread the line, I've made the mistake so many times that I see it everywhere now. But the fact that the operands are switched makes it harder to read somehow.

Reply:
As a Python programmer, I'm used to saying "value X is between A and B" with `A <= X <= B` but of course that doesn't work in C++.

Now that I look, I see `unicode.hh` has an `is_basic_digit()` function that's basically that, so I guess I should use it.